packages/rsocket-ws-gateway/src/createGatewayProxy.ts

```diff
@@ -3,8 +3,14 @@ import RSocketWebSocketClient from 'rsocket-websocket-client';
 import { RSocketClient, JsonSerializers } from 'rsocket-core';
 import { unpackError } from './utils';
 
-export const createGatewayProxy = (url, definition) => {
-  const proxy = {};
+type createGatewayProxyType = (url: string, definitions: any) => Promise<any>;
```

Comment:
please put it in types folder
and please describe `definitions` better
also you can add:
```ts
type Proxy = any;
```
but i don't think its possible to promise a proxy, you will have to do like this:

```suggestion
type createGatewayProxyType = (url: string, definitions: ServiceDefinition) => Promise<Proxy[]>| Promise<Proxy>;
```